src/tests/system/tests/test_kcm.py

```diff
@@ -413,3 +415,335 @@ def test_kcm__simple_kinit(client: Client, kdc: KDC):
+        5. Remove kcm log files
+        6. Start SSSD
+    :steps:
+        1. Try to produce some debug messages e.g. kdestroy
+        2. Check that kcm debug messages were not generated
+        3. Set "debug_level" in kcm config section to "9"
+        4. Restart kcm
+        5. Try to produce some debug messages e.g. kdestroy
+        6. Check that kcm debug messages were generated
+    :expectedresults:
+        1. No messages were generated
+        2. Log file did not get bigger
+        3. Successfully set
+        4. Successfully restarted
+        5. Some messages were generated
+        6. Log file did get bigger
+    :customerscenario: False
+    """
+
+    def kcm_log_length() -> int:
+        try:
+            output = client.fs.wc(kcm_log_file, lines=True).stdout
+            return int(output.split()[0])
+        except SSHProcessError:
+            return 0
+
+    user = "user1"
+    password = "Secret123"
+    kcm_log_file = "/var/log/sssd/sssd_kcm.log"
+
+    kdc.principal(user).add(password=password)
+    client.local.user(user).add(password=password)
+    client.sssd.common.kcm(kdc)
+
+    client.sssd.kcm["debug_level"] = "0"
+    client.sssd.config_apply()
+
+    client.ssh(user, password).exec(["rm", "-f", kcm_log_file], raise_on_error=False)
+    client.sssd.start()
+
+    start_log_length = kcm_log_length()
+
+    with client.ssh(user, password) as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            krb.kdestroy()
+
+    end_log_nodebug = kcm_log_length()
+    assert start_log_length == end_log_nodebug, "Debug messages were generated"
+
+    client.sssd.kcm["debug_level"] = "9"
+    client.sssd.config_apply()
+    assert client.svc.restart("sssd-kcm").rc == 0, "Restart of kcm failed"
+
+    with client.ssh(user, password) as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            krb.kdestroy()
+
+    end_log_debug = kcm_log_length()
+    assert start_log_length + 100 < end_log_debug, "Debug messages were not generated"
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_kcm__ssh_login_with_kcm(client: Client, ldap: LDAP, kdc: KDC):
+    """
+    :title: kcm: Verify ssh login is successuful with kcm as default
+    :setup:
+        1. Add user and principal
+        2. Set kerberos as default auth provider
+        3. Start SSSD
+    :steps:
+        1. Authenticate as "user1" over SSH using kcm
+    :expectedresults:
+        1. Authenticated successfully
+    :customerscenario: False
+    """
+    ldap.user("user1").add()
+    kdc.principal("user1").add()
+
+    client.sssd.common.krb5_auth(kdc)
+    client.sssd.start()
+
+    with client.ssh("user1", "Secret123") as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            res = krb.klist()
+            assert res.rc == 0, "Klist failed"
+
+
+@pytest.mark.ticket(gh=4763)
+@pytest.mark.topology(KnownTopology.Client)
+def test_kcm__display_correct_kvno(client: Client, kdc: KDC):
```

Comment:
This test case can be removed, it's covered by "test_kcm__ccache_holds_multiple_and_all_types_of_principals", the title hasn't been merged yet, but you can compare it from the audit PR.